### ISSUE TITLE
Build fails in Travis CI and locally in macOS XCode 9.2 environment for Flutter plugins

### ISSUE THREAD

**Issue by alibitek** [flutter/flutter#13813]
## Steps to Reproduce

[Flutter plugin repo](https://github.com/flutter/plugins) fails to build in Travis CI - osx xcode9.2 env - due to`pod install --verbose` failure when building IPA file for the first plugin: [packages/android_alarm_manager/example](https://github.com/flutter/plugins/tree/master/packages/android_alarm_manager/example)

[The command executed is](https://travis-ci.org/flutter/plugins/jobs/323172152#L874): `./script/plugin_tools.sh build-examples --ipa`

- [Job Config](https://travis-ci.org/flutter/plugins/jobs/323172152/config)
- [Job Logs](https://travis-ci.org/flutter/plugins/jobs/323172152#L1307)

```
Warning: Building for device with codesigning disabled. You will have to manually codesign before deploying to device.
Building io.flutter.androidAlarmManagerExample for device (ios-release)...
Upgrading project.pbxproj of Runner.app' to include the 'flutter_assets' directory
Removing app.flx from project.pbxproj since it has been replaced with flutter_assets.
Running pod install...
CocoaPods' output:
↳
    Invalid plugin specification: // This is a generated file; do not edit or check into version control.
    Invalid plugin specification: #include "Pods/Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"
...
    [!] The 'Pods-Runner' target has transitive dependencies that include static binaries: 
(/Users/travis/build/flutter/plugins/packages/android_alarm_manager/example/ios/Pods/GoogleSignIn/Frameworks/GoogleSignIn.framework)
...

The following build are failing (see above for details):
 * android_alarm_manager/example (ipa)
The command "./script/plugin_tools.sh build-examples --ipa" exited with 1.
```

Latest commit on Flutter `master branch` from PR: https://github.com/flutter/flutter/pull/13774 changed [Obj-C and Swift CocoaPods's Podfile templates](https://github.com/flutter/flutter/tree/master/packages/flutter_tools/templates/cocoapods): https://github.com/flutter/flutter/commit/4d2c2aaaa175e2d501ecadaa767348db563f73af#diff-51b9cab5bac9e12013d7a4391d6ac7a1

by adding `use_frameworks!` which causes problems in CocoaPods 1.3.1:
- https://github.com/flutter/flutter/issues/10968
- https://github.com/flutter/flutter/issues/12818
- https://github.com/flutter/flutter/pull/10205
- https://github.com/flutter/flutter/issues/9694

**Comment by mravn-google** [comment]
cc @xster 

I've fixed the `android_alarm_manager` plugin by checking in a `Podfile` without the `use_frameworks!` line.

But #13774 seems to have introduced a general problem:

1. Switch Flutter to use the master branch
1. `flutter create tst11`; `cd tst11`
1. Add dependency on `google_sign_in` to `pubspec.yaml`
1. `flutter build ios`

Then
```text
$ flutter build ios
Building com.yourcompany.tst11 for device (ios-release)...
Found saved certificate choice "iPhone Developer: Mikkel Ravn (XXXXXXXXXX)". To clear, use "flutter config".
Signing iOS app for device deployment using developer identity: "iPhone Developer: Mikkel Ravn (XXXXXXXXXX)"
Running pod install...                                0.8s
CocoaPods' output:
↳
    Invalid plugin specification: // This is a generated file; do not edit or check into version control.
    Invalid plugin specification: #include "Pods/Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"
      Preparing
    
    Analyzing dependencies
    
    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (`arm64`)
    
    Fetching external sources
    -> Fetching podspec for `Flutter` from `/Users/mravn/github/flutter/bin/cache/artifacts/engine/ios-release`
    -> Fetching podspec for `google_sign_in` from `/Users/mravn/.pub-cache/hosted/pub.dartlang.org/google_sign_in-2.0.0/ios`
    
    Resolving dependencies of `Podfile`
    
    Comparing resolved specification to the sandbox manifest
      A Flutter
      A GTMOAuth2
      A GTMSessionFetcher
      A GoogleSignIn
      A GoogleToolboxForMac
      A google_sign_in
    
    Downloading dependencies
    
    -> Installing Flutter (1.0.0)
    
    -> Installing GTMOAuth2 (1.1.5)
      > Copying GTMOAuth2 from `/Users/mravn/Library/Caches/CocoaPods/Pods/Release/GTMOAuth2/1.1.5-be83f` to `Pods/GTMOAuth2`
    
    -> Installing GTMSessionFetcher (1.1.12)
      > Copying GTMSessionFetcher from `/Users/mravn/Library/Caches/CocoaPods/Pods/Release/GTMSessionFetcher/1.1.12-ebaa1` to `Pods/GTMSessionFetcher`
    
    -> Installing GoogleSignIn (4.1.1)
      > Copying GoogleSignIn from `/Users/mravn/Library/Caches/CocoaPods/Pods/Release/GoogleSignIn/4.1.1-3fdaa` to `Pods/GoogleSignIn`
    
    -> Installing GoogleToolboxForMac (2.1.3)
      > Copying GoogleToolboxForMac from `/Users/mravn/Library/Caches/CocoaPods/Pods/Release/GoogleToolboxForMac/2.1.3-2501e` to `Pods/GoogleToolboxForMac`
    
    -> Installing google_sign_in (0.0.1)
      - Running pre install hooks
    [!] The 'Pods-Runner' target has transitive dependencies that include static binaries: (/Users/mravn/apps/tst11/ios/Pods/GoogleSignIn/Frameworks/GoogleSignIn.framework)
    
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:434:in `block (2 levels) in verify_no_static_framework_transitive_dependencies'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:426:in `each'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:426:in `block in verify_no_static_framework_transitive_dependencies'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:423:in `each'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:423:in `verify_no_static_framework_transitive_dependencies'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/installer.rb:113:in `install!'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/command/install.rb:41:in `run'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/lib/cocoapods/command.rb:52:in `run'
    /Users/mravn/.rvm/gems/ruby-2.2.2/gems/cocoapods-1.2.1/bin/pod:55:in `<top (required)>'
    /Users/mravn/.rvm/gems/ruby-2.2.2/bin/pod:23:in `load'
    /Users/mravn/.rvm/gems/ruby-2.2.2/bin/pod:23:in `<main>'
    /Users/mravn/.rvm/gems/ruby-2.2.2/bin/ruby_executable_hooks:15:in `eval'
    /Users/mravn/.rvm/gems/ruby-2.2.2/bin/ruby_executable_hooks:15:in `<main>'
Error output from CocoaPods:
↳
    
    [!] Automatically assigning platform ios with version 8.0 on target Runner because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.
Error running pod install
```

**Comment by mravn-google** [comment]
Just noticed that we already have a PR to fix this: #13814.
